src/development/dev-tests-coding.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: article
+title: "Writing clean tests"
```

Comment:
Can we include a section on testing for exceptions?

Testing shouldn't just cover happy path but the tests should specifically handle the exceptions they produce.  I see a lot of test output littered with stack traces, which doesn't feel right. It makes me worried these tests are passing through false positives.